src/app/authentication/service/signUpService.go

```diff
@@ -1 +1,52 @@
 package service
+
+import (
+	"authentication/constants"
+	"authentication/models"
+	"authentication/repo"
+	"authentication/utils"
+	"errors"
+)
+
+type UserRepo struct {
+	userExistenceChecker repo.UserExistenceChecker
+}
+
+func NewUserRepo(checker repo.UserExistenceChecker) *UserRepo {
+	return &UserRepo{
+		userExistenceChecker: checker,
+	}
+}
+
+// @Summary Register a new customer
+// @Description Registers a new customer and saves their data in the database.
+// @Tags customers
+// @Accept json
+// @Produce json
+// @Param customer body models.Customer true "Customer data"
+// @Success 200 {string} string "success"
+// @Failure 400 {object} string "Bad Request"
+// @Failure 409 {object} string "Email already exists"
+// @Failure 409 {object} string "Phone number already exists"
+// @Failure 409 {object} string "Pan card number already exists"
+// @Failure 422 {object} string "Invalid customer data"
+// @Failure 500 {object} string "Internal Server Error"
```

Comment:
All Swagger codes should only be written in the handler file. Implement all Failure Swaggers in the handler file itself.